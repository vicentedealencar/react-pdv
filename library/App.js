import React from 'react'
import { Library, Example } from '@compositor/kit'
import * as examples from './examples'
import { ThemeProvider } from 'styled-components'

import theme from './theme'

export default () => (
  <ThemeProvider theme={theme}>
    <Library title="react-pdv">
      {Object.entries(examples).map(([name, Component]) => (
        <Example name={name} key={name}>
          <Component />
        </Example>
      ))}
    </Library>
  </ThemeProvider>
)
