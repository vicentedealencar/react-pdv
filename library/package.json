{
  "private": true,
  "name": "react-pdv-library",
  "description": "react-pdv component library",
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "format": "prettier --write *.js"
  },
  "devDependencies": {
    "prettier": "^1.12.1",
    "react-scripts": "^1.1.4"
  },
  "prettier": {
    "semi": false,
    "singleQuote": true
  },
  "dependencies": {
    "@compositor/kit": "^1.0.47",
    "react": "^16.4.1",
    "react-dom": "^16.4.1",
    "react-pdv": "0.1.4",
    "standard-components": "^1.0.0-2"
  },
  "version": "1.0.0"
}
