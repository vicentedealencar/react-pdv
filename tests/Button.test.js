import snapshotExamples from '@compositor/kit-snapshot'
import * as examples from '../library/examples'

snapshotExamples(examples)
