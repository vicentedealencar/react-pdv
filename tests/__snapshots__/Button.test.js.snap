// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicButton renders 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#770077",
      "border": "5px solid black",
      "borderRadius": 20,
      "color": "#fff",
      "paddingBottom": 10,
      "paddingLeft": 100,
      "paddingRight": 100,
      "paddingTop": 10,
    }
  }
>
  I am a button, please click me
</button>
`;

exports[`Cart renders 1`] = `
<div
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#ebebef",
      },
      Object {
        "backgroundColor": "#770000",
        "border": "5px solid black",
        "borderRadius": 20,
        "color": "#fff",
        "paddingBottom": 10,
        "paddingLeft": 100,
        "paddingRight": 100,
        "paddingTop": 10,
      },
    ]
  }
>
  <div>
    <p
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 28,
          "fontWeight": "900",
          "marginBottom": 30,
          "marginTop": 30,
        }
      }
    >
      Total: 802.35
    </p>
    <div
      style={
        Object {
          "borderBottomWidth": 4.5,
          "borderColor": "#d6d7da",
        }
      }
    >
      <div>
        <p>
          prod1
           - 
          123.45
        </p>
      </div>
      <div>
        <p>
          prod2
           - 
          678.9
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`LoadingButton renders 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#770077",
      "border": "5px solid black",
      "borderRadius": 20,
      "color": "#fff",
      "paddingBottom": 10,
      "paddingTop": 10,
      "textAlign": "center",
      "width": 300,
    }
  }
>
  Delayed counter: 0
</button>
`;
